packages/react-reconciler/src/ReactFiberScheduler.js

```diff
@@ -876,7 +876,8 @@ export default function<T, P, I, TI, HI, PI, C, CC, CX, PL>(
     } while (true);
 
     // We're done performing work. Time to clean up.
-    stopWorkLoopTimer(interruptedBy);
+    const didCompleteRoot = nextUnitOfWork === null && isRootReadyForCommit;
+    stopWorkLoopTimer(interruptedBy, didCompleteRoot);
```

Comment:
Nit: could you move this into the branches below? That way it's less likely they get out of sync.